### ISSUE TITLE
custom configuration via docker-compose.yml

### ISSUE THREAD

**Issue by b3nk3** [jwilder/nginx-proxy#507]
Hi, 

I am a bit unsure on how to implement this:
```
FROM jwilder/nginx-proxy
RUN { \
      echo 'server_tokens off;'; \
      echo 'client_max_body_size 100m;'; \
    } > /etc/nginx/conf.d/my_proxy.conf
```
In my compose file? I don't necesserly need a full custom config file, only the `client_max_body_size` part

**Comment by wader** [comment]
One way is to put that in a local Dockerfile next to your docker-compose.yml and use `build: .` in docker-compose.yml. Then do `docker-compose build` before up.

**Comment by b3nk3** [comment]
I see, i don't really want to mess about with own dockerfiles...

So how would I add my custom config file? As a volume?
`/host/path/to/custom/nginx.conf:/etc/nginx/conf.d/my_proxy.conf`

**Comment by wader** [comment]
Hmm yeah that i think should work too... actually there is a example for doing that in README.md

**Comment by b3nk3** [comment]
I know, but that involves a dockerfile as well. I just want to keep it the simplest, compose only.

**Comment by wader** [comment]
Sorry i was referring to the example in READMe.md for using a volume for custom config
```sh
docker run -d -p 80:80 -p 443:443 -v /path/to/my_proxy.conf:/etc/nginx/conf.d/my_proxy.conf:ro -v /var/run/docker.sock:/tmp/docker.sock:ro jwilder/nginx-proxy
```
For compose that should just be a `- volumes:` line.

**Comment by b3nk3** [comment]
Right, added my .conf as a volume, it works but then I get this error:

`[emerg] 11#11: "daemon" directive is not allowed here in /etc/nginx/conf.d/nginx.conf:1`

Any clues?

**Comment by wader** [comment]
Hmm how does `/etc/nginx/conf.d/nginx.conf` look? note that `/etc/nginx/nginx.conf` has this:
```
http {
    ... cut ...
    include /etc/nginx/conf.d/*.conf;
}
daemon off;
```

So if there is a `daemon` directive in `/etc/nginx/conf.d/nginx.conf` is will be inside `http { ... } which would give that error. Try strip out all global config from your config and only keep the http stuff if that is possible.

**Comment by theiThomas** [comment]
Hey,
How do you do this ? I want to do the same.

Thanks
